ale/formatters/usgscsm_formatter.py

```diff
@@ -81,21 +82,57 @@ def to_usgscsm(driver):
     if isinstance(driver, LineScanner):
         isd_data['name_model'] = 'USGS_ASTRO_LINE_SCANNER_SENSOR_MODEL'
         isd_data['interpolation_method'] = 'lagrange'
+
         start_lines, start_times, scan_rates = driver.line_scan_rate
         center_time = driver.center_ephemeris_time
         isd_data['line_scan_rate'] = [[line, time, rate] for line, time, rate in zip(start_lines, start_times, scan_rates)]
         isd_data['starting_ephemeris_time'] = driver.ephemeris_start_time
         isd_data['center_ephemeris_time'] = center_time
-        isd_data['t0_ephemeris'] = position_times[0] - center_time
-        if len(position_times) > 1:
-            isd_data['dt_ephemeris'] = (position_times[-1] - position_times[0]) / (len(position_times) - 1)
+
+        interp_times = np.linspace(position_times[0],
+                                   position_times[-1],
+                                   int(driver.image_lines / 64))
+
+        if velocities:
+            positions = np.asarray(positions)
+            velocities = np.asarray(velocities)
+            x_interp = BPoly.from_derivatives(position_times,
+                                              np.vstack((positions.T[0], velocities.T[0])).T,
+                                              extrapolate=True)
+            y_interp = BPoly.from_derivatives(position_times,
+                                              np.vstack((positions.T[1], velocities.T[1])).T,
+                                              extrapolate=True)
+            z_interp = BPoly.from_derivatives(position_times,
+                                              np.vstack((positions.T[2], velocities.T[2])).T,
+                                              extrapolate=True)
```

Comment:
these transposes suck, is there a better way to do this? Basically this function wants [[pos_0, vel_o],[pos_1, vel1],...,[pos_n, vel_n]]

Reply:
you can reduce to one transpose for position and orientation:
```
px,py,pz = np.asarray(positions).T
vx,vy,vz = np.asarray(velocities).T
```

that would make things a bit cleaner.